### ISSUE TITLE
Dean's metering-limit crash

### ISSUE THREAD

**Issue by warner** [Agoric/agoric-sdk#892]
@dtribble told us he saw a cosmic-swingset crash with a metering-limit error yesterday. He said he'd done a few normal contract things, and then left it idle overnight. The next morning it had halted.

We're guessing that something (triggered by a timer) is running wild, maybe a sorcerors-apprentice -type bug or `O(n^2)` kind of thing. The first step will be to add a diagnostic that displays some basic info on each block, something like:

* number of timer events fired
* number of cranks run
* number of syscalls made

**Comment by dtribble** [comment]
Here's a snippet of the log that shows the exception: 
```
launch-chain: mboxDeliver:   ADDED messages
SwingSet: kernel: doProcess: vat[v9][o+4].wake dispatch failed: RangeError: Compute meter exceeded
    at RangeError (eval at <anonymous> (eval at createScopedEvaluatorFactory (/mnt/c/dev/git/agoric-sdk/node_modules/ses/dist/ses.cjs.js:1323:10)), <anonymous>:111:20)
    at Object.eval [as c] (/SwingSet/vat-zoe/agoric-sdk/packages/transform-metering/src/meter.js:87:26)```

**Comment by dtribble** [comment]
Current theory is that it is the timer activity from `zoeNotifier`.  The code uses:
```js
  const POLL_DELAY_S = 4;
  ...
  const repeater = E(timerService).createRepeater(0, POLL_DELAY_S);
  E(repeater).schedule(timerHandler);
```

**Comment by dtribble** [comment]
I got a further issue when polling the ag-solo. Is this a meter exhaustion on ag-solo?
```
SwingSet: vat-http: Error onMessage: RangeError: Compute meter exceeded
    at fullRevive (/SwingSet/kernel/packages/marshal/marshal.js:615:50)
    at Object.unserialize (/SwingSet/kernel/packages/marshal/marshal.js:663:21)
    at Object.notifyReject (/SwingSet/kernel/packages/SwingSet/src/kernel/liveSlots.js:438:21)
    at process (/SwingSet/kernel/packages/SwingSet/src/kernel/vatManager.js:298:48)
^Coutbound: deliver { 'fake-chain': { outbox: [ [Array] ], inboundAck: 53 } }
web: 127.0.0.1:58602[2]: inbound error: Compute meter exceeded from {
  "type": "dvp/getBalance",
  "meta": {
    "channelID": 2,
    "origin": "http://localhost:3000",
    "url": "/api",
    "date": 1586409864014
  }
}```

**Comment by warner** [comment]
That certainly looks like a meter exhaustion message.

I've been unable to reproduce this so far.

I ran what I think was Dean's setup: commit 6c449fc3765472297a17c2f1949d06f222354e6d, ran the Getting Started instructions, performed one Simple Exchange, left everything running. I modified the timers to produce blocks faster (about ten blocks per second at the start, with the timer-device "wake" interval producing about one wakeup per second). And I left it all running for 54 hours without seeing a meter overflow. I'm not convinced that any metered code was actually running all that time, I still don't understand enough to know what should have been subject to the meter and when it got run.

This commit is before the #899 switch to DB-backed kernel storage, so it keeps the primary state in a `Map` (in RAM) and writes it out to a `.jsonlines` file after each block. Within 3-4 hours of startup, the time it took to serialize the `Map` grew large enough that the CPU was no longer idle: CPU usage rose from a few percent to 100%. After that point, the block creation rate slowly dropped. At the end of 54 hours it was taking maybe 5s to produce each block.

Memory usage on a swap-enabled virtual-memory system is always pretty nebulous, but I saw the process' VmSize at 1.36GB and growing slowly, and the RSS hit 512MB (and growing at about 1.2MB per minute). The ag-solo `.jsonlines` file was growing about 1 key per block, while the fake-chain file was growing by about 11 keys per block. I believe the only activity at this point was a timer-device wakeup every block, which was delivered through the comms vat to the ag-solo machine, whereupon it was dispatched into a vat and then ignored.

Then I shut everything down and updated to a post-#899 commit (6d107befde7c0955aaf1014126bc1bc97e08203d) and ran the test again, this time for 15 hours. As expected the performance was much better: I was still getting about 10 blocks/second at the end of the 15 hours, and the CPU usage remained constant at 4.5% of a core. The fake-chain database file was growing at about 2650 bytes per second (so maybe 265 bytes per block), and the ag-solo DB was about 179 bytes per block. Curiously the VmSize was much larger to begin with, 5.26GB, but remained fairly constant throughout. The RSS grew steadily, by about 4270 bytes per second (so maybe 427 bytes per block), of which some portion must be the result `Promise` objects for the `wake` calls, which should be somewhat mitigated by Chip's  upcoming promise-retirement work.
